src/live/handleAlbumChat.js

```diff
@@ -1,11 +1,36 @@
-function handleAlbumChat(io, socket, data) {
-  const { message, userId, albumId } = data;
+const ChatMessage = require('../models/ChatMessage');
 
-  // Create a chat room name for the album chat
-  const chatRoomName = `chat:album:${albumId}`;
+async function handleAlbumChat(io, socket, data) {
+  try {
+    const { albumId } = data;
 
-  // Send the message to the room
-  socket.to(chatRoomName).emit('chat:album', { message, userId, albumId });
+    // Create a chat room name for the album chat
+    const chatRoomName = `chat:album:${albumId}`;
+
+    const msg = await ChatMessage.create({ ...data, name: data.user });
+    await msg.populate('user', 'name');
+    console.log('Chat message saved:', msg);
+
+    const savedMessage = {
+      ...data,
+      _id: msg._id, // Include _id in the message
+      name: data.name,
+    };
+
+    // Check and delete old messages if needed
+    const messagesCount = await ChatMessage.countDocuments({ albumId });
+    if (messagesCount > 50) {
+      await ChatMessage.findOneAndDelete(
+        { albumId },
+        { sort: { timestamp: 1 } }
+      );
+    }
+
+    // Broadcast the message to all connected users in the album chat room
+    io.to(chatRoomName).emit('chat:album', savedMessage);
```

Comment:
My initial suggestion was to return `msg` since that will have all the info we need and any future data we add to `ChatMessage` schema. But this works as well, where you're sending a mix of info from the original socket payload (`data`) and your chosen subset of fields from the database `msg` (in your case, just `_id`). :+1: